alternator/streams.cc

```diff
@@ -932,15 +947,8 @@ future<executor::request_return_type> executor::get_records(client_state& client
         for (auto& row : result_set->rows()) {
             auto op = static_cast<cdc::operation>(value_cast<op_utype>(data_type_for<op_utype>()->deserialize(*row[op_index])));
             auto ts = value_cast<utils::UUID>(data_type_for<utils::UUID>()->deserialize(*row[ts_index]));
-
-            if (timestamp && timestamp != ts) {
-                maybe_add_record();
-                if (limit == 0) {
-                    break;
-                }
-            }
-
-            timestamp = ts;
+            // end_of_batch is boolean, but always null or true.
+            auto eor = row[eor_index].has_value();
```

Comment:
Just for the sake of being save in the future, I would check the value anyway in case we change the behavior and the false value actually starts appearing.